modules/programs/info.nix

```diff
@@ -60,7 +60,7 @@ in
     }];
 
     home.sessionVariables.INFOPATH =
-      "${cfg.homeInfoDirLocation}\${INFOPATH:+:}\${INFOPATH}";
+      "${cfg.homeInfoDirLocation}\$INFOPATH:+:\$INFOPATH";
```

Comment:
I'm suspecting this situation is common enough that we could have
```
home.sessionVariables.*.set
home.sessionVariables.*.append
home.sessionVariables.*.prepend
home.sessionVariables.*.separator
```
with `home.sessionVariables.XYZ.set = "foo"` interpreted as
```
home.sessionVariables.XYZ = {
  set = "foo";
  append = null;
  prepend = null;
  separator = null;
}
```

And
```
home.sessionVariables.INFOPATH = "${cfg.homeInfoDirLocation}\${INFOPATH:+:}\${INFOPATH}";
```
would then be written
```
home.sessionVariables.INFOPATH = {
  append = cfg.homeInfoDirLocation;
  separator = ":";
};
```
or
```
home.sessionVariables.INFOPATH.append = cfg.homeInfoDirLocation;
```
if we default `separator` to ":".

This should cover the majority of cases, I think. It would also be possible to add
```
home.sessionVariables.*.rawPamValue
home.sessionVariables.*.rawCshValue
home.sessionVariables.*.rawBshValue
```
that would be set automatically by the above options but could be set directly if a user wants something advanced that is not possible otherwise.